src/main/java/org/mate/bookstorespringboot/service/impl/BookServiceImpl.java

```diff
@@ -33,8 +34,33 @@ public List<BookDto> findAll() {
 
     @Override
     public BookDto findById(Long id) {
-        Book book = bookRepository.findById(id)
-                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
-        return bookMapper.toDto(book);
+        return bookRepository.findById(id)
+                .map(bookMapper::toDto)
+                .orElseThrow(() -> new EntityNotFoundException(BOOK_NOT_FOUND + id));
+    }
+
+    @Override
+    public BookDto updateBook(Long id, BookRequestDto bookRequestDto) {
+        checkIfBookExists(id);
+        Book updatedBook = bookRepository.findById(id)
+                .map(existingBook -> {
```

Comment:
You can combine checking and receiving entities from DB. Just return Optional and use isPresent()